packages/sdk/src/modules/FusePools.ts

```diff
@@ -96,13 +99,25 @@ export function withFusePools<TBase extends FuseBaseConstructor>(Base: TBase) {
         asset.totalBorrowNative =
           Number(utils.formatUnits(asset.totalBorrow)) * Number(utils.formatUnits(asset.underlyingPrice));
 
+        if (asset.totalSupplyNative === 0) {
+          asset.utilization = 0;
+        } else {
+          asset.utilization = (asset.totalBorrowNative / asset.totalSupplyNative) * 100;
+        }
+
         totalSuppliedNative += asset.totalSupplyNative;
         totalBorrowedNative += asset.totalBorrowNative;
 
         asset.liquidityNative =
           Number(utils.formatUnits(asset.liquidity)) * Number(utils.formatUnits(asset.underlyingPrice));
 
         totalLiquidityNative += asset.liquidityNative;
+
+        if (!asset.isSupplyPaused) suppliedForUtilization += asset.totalSupplyNative;
+        if (!asset.isBorrowPaused) borrowedForUtilization += asset.totalBorrowNative;
+        if (suppliedForUtilization !== 0) {
+          utilization = (borrowedForUtilization / suppliedForUtilization) * 100;
```

Comment:
This calculation can be done outside of the for loop.

Reply:
But maybe this PR is more about moving hte calculation outside the FE and into SDK and not about this ticket here yet, right?

https://github.com/Midas-Protocol/monorepo/issues/184

Reply:
Yes but also this PR fixed `utilization` calculation based on only borrow-able assets

Reply:
makes sense

Reply:
> Also not sure if this calculation is correct. thinking
> 
> Shouldn't it be something like `currently borrowed / total amount which is abled to borrowed` ?
> 
> Where as `total amount which is abled to borrowed` is `sum of all the supply of borrowable assets`, so not including the supply of assets which are not borrowable.
> 
> `currently borrowed` => `sum of all borrowed assets`

Check this https://github.com/Midas-Protocol/monorepo/pull/201/commits/a965ecbd72e6a7a9c5f866f261e5004f600463c5

Reply:
Perfect, I think this should work as intended in #184